src/Client/SSHConnection.ts

```diff
@@ -40,6 +45,30 @@ export const createSSHConnection: ConnectionFactory = async (host, port) => {
 	await once(conn, 'ready')
 	return {
 		async run(streams: ProcessStreams, job: Job): Promise<JobResult> {
+			// TBD Transfter files to erv s jobstopushinputreq
+			// TBD  peipe tar.c to stdin of exec
+			// We send a put request to the daemon
+			const putPayload = JSON.stringify(jobToPushInputs(job))
+			const pushInputStream = await promisify(conn.exec.bind(conn))(putPayload)
+			// We generate list of jobs to be tarred in the stream.
+			const arrayOfJobs: Array<Job> = job.getDeepPrerequisitesIterable()
+			const fileList: Array<string> = []
+			for (const subJob of arrayOfJobs) {
+				fileList.push(subJob.getName())
+			}
```

Comment:
Should you also be calling `getPrerequisiteFilesIterable()` here? The `Job` might need files that aren't targets.

Reply:
I asked @rvighne for clarification regarding this. It is my understanding that the job name is all we need.

Reply:
Yeah I was also under the impression that job name = target name in the Makefile = file name on disk

Reply:
Yes, but what about a preexisting file `a.c` that needs to be there from the start, instead of being built during the process?

Reply:
I see, yeah that's important. @pjot726 can you make sure you take care of this situation?

As I understand it, if we have
```makefile
intermediate.txt: input.txt
    ...

output.txt: intermediate.txt
    ...
```
there will be `Job`s created for `output.txt` and `intermediate.txt`, but not `input.txt`, and to handle that we need the function Rohan recommended.

Reply:
Latest push should address this issue